src/hotspot/share/gc/shenandoah/shenandoahFreeSet.cpp

```diff
@@ -111,163 +453,135 @@ HeapWord* ShenandoahFreeSet::allocate_single(ShenandoahAllocRequest& req, bool&
         return nullptr;
       }
 
-      // Try to steal the empty region from the mutator view
-      for (size_t c = _mutator_rightmost + 1; c > _mutator_leftmost; c--) {
+      // Try to steal an empty region from the mutator view.
+      for (size_t c = _partitions.rightmost_empty(Mutator) + 1; c > _partitions.leftmost_empty(Mutator); c--) {
```

Comment:
`ssize_t` to keep all these loops uniform.

Reply:
Agree.  Done.